plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/XBuildManager.java

```diff
@@ -247,27 +250,28 @@ private XBuildable doGetIncrementalBuildable(List<URI> dirtyFiles, List<URI> del
 		try {
 			Map<ProjectDescription, Set<URI>> project2dirty = computeProjectToUriMap(this.dirtyFiles);
 			Map<ProjectDescription, Set<URI>> project2deleted = computeProjectToUriMap(this.deletedFiles);
+			SetView<ProjectDescription> changedPDs = Sets.union(project2dirty.keySet(), project2deleted.keySet());
+			IOrderInfo<ProjectDescription> orderInfo = projectBuildOrderProvider.get(changedPDs);
```

Comment:
How much time does it take to compute the build order for OPR?

Reply:
I measured the method ProjectOrderInfo#init() from top to bottom with a Stopwatch. The output of the LSP server (started with `--clean`) on OPR initial build was:

```
LSP server starts
Listening for LSP clients on port 5007...
Connected to LSP client
LSP Server connected
8.379 ms
6.776 ms
```

So the build order was computed twice. First for the initial build, and second for an incremental build when the server sent the openFile call, I guess ("I guess", because the latter number `6.776 ms` was printed out after the initial build was complete).